Source/Core/loadKTX2.js

```diff
@@ -0,0 +1,114 @@
+import arraySlice from "./arraySlice.js";
+import Check from "./Check.js";
+import defined from "./defined.js";
+import Resource from "./Resource.js";
+import RuntimeError from "./RuntimeError.js";
+import KTX2Transcoder from "../Scene/KTX2Transcoder.js";
+import when from "../ThirdParty/when.js";
+
+/**
+ * Stores the supported formats that KTX2 can transcode to. Called during context creation.
+ *
+ * @param {Boolean} s3tc Whether or not S3TC is supported
+ * @param {Boolean} pvrtc Whether or not PVRTC is supported
+ * @param {Boolean} astc Whether or not ETC1 is supported
+ * @param {Boolean} etc1 Whether or not ASTC is supported
+ * @param {Boolean} bc7 Whether or not BC7 is supported
+ * @private
+ */
+var supportedTranscoderFormats;
+
+loadKTX2.setKTX2SupportedFormats = function (s3tc, pvrtc, astc, etc1, bc7) {
+  supportedTranscoderFormats = {
+    s3tc: s3tc,
+    pvrtc: pvrtc,
+    astc: astc,
+    etc1: etc1,
+    bc7: bc7,
+  };
+};
+
+/**
+ * Asynchronously loads and parses the given URL to a KTX2 file or parses the raw binary data of a KTX2 file.
+ * Returns a promise that will resolve to an object containing the image buffer, width, height, and format once loaded,
+ * or reject if the URL failed to load or failed to parse the data. The data is loaded
+ * using XMLHttpRequest, which means that in order to make requests to another origin,
+ * the server must have Cross-Origin Resource sharing (CORS) headers enabled.
+ * <p>
+ * The following are part of the KTX2 format specification but are not supported:
+ * <ul>
+ *     <li>Metadata</li>
+ *     <li>3D textures</li>
+ *     <li>Texture Arrays</li>
+ *     <li>Video</li>
+ *     <li>Compressed Mipmaps</li>
+ * </ul>
+ * </p>
+ *
+ * @exports loadKTX2
+ *
+ * @param {Resource|String|ArrayBuffer} resourceOrUrlOrBuffer The URL of the binary data or a ArrayBuffer.
+ * @returns {Promise.<CompressedTextureBuffer>|undefined} A promise that will resolve to the requested data when loaded. Returns undefined if <code>request.throttle</code> is true and the request does not have high enough priority.
+ *
+ * @exception {RuntimeError} Invalid KTX2 file.
+ * @exception {RuntimeError} File is the wrong endianness.
+ * @exception {RuntimeError} glInternalFormat is not a valid format.
+ * @exception {RuntimeError} glType must be zero when the texture is compressed.
+ * @exception {RuntimeError} The type size for compressed textures must be 1.
+ * @exception {RuntimeError} glFormat must be zero when the texture is compressed.
+ * @exception {RuntimeError} Generating mipmaps for a compressed texture is unsupported.
+ * @exception {RuntimeError} The base internal format must be the same as the format for uncompressed textures.
+ * @exception {RuntimeError} 3D textures are not supported.
+ * @exception {RuntimeError} Texture arrays are not supported.
+ * @exception {RuntimeError} Cubemaps are not supported.
```

Comment:
they are now - between these and the specs all the possible error messages are covered. I'm not sure what would cause the `startTranscoding()` and `transcodeImage()` failures, but the Binomial example has those in their [WebGL demo]https://github.com/BinomialLLC/basis_universal/blob/c4c0db735a6a799702cd4e7309df64b4f7f59e39/webgl/ktx2_encode_test/index.html#L393)..

Reply:
Like I mentioned in my review comment on loadKTX2Spec, I'm not sure what level or granularity we want with these errors.

Reply:
This level of granularity is fine